Amazon.IonObjectMapper.Test/TestObjects.cs

```diff
@@ -292,10 +292,43 @@ public override string ToString()
 
     public class Wheel
     {
+        public string Brand { get; init; }
```

Comment:
Can we create an object with an Ion constructor and with object parameters. I wonder if the current logic handles this.